### ISSUE TITLE
"Please check validity" for no reason (?) when defining mixin with "when"

### ISSUE THREAD

**Issue by adam-lynch** [lesshint/lesshint#133]
```less
.something(@i: 1) when (@i =< 4) {
  //...
}
```

>parse error: Please check validity of the block starting from line...

**Comment by jwilsson** [comment]
Ah, the good ol' parsing issues. Our parser, `gonzales-pe` has some issues with the Less syntax. @shellscape has done a great job of fixing some of it in our fork (https://github.com/gilt/gonzales-pe) as well as working a PostCSS Less parser (https://github.com/gilt/postcss-less) and he will probably love you if you have the time/want to help out with some of it :)

**Comment by adam-lynch** [comment]
Will see what I can do.. :smile:

**Comment by screeny05** [comment]
+1

**Comment by miloszsobczak** [comment]
+1

**Comment by shellscape** [comment]
FYI this was reported to gonzales-pe in January https://github.com/tonyganch/gonzales-pe/issues/146

**Comment by adam-lynch** [comment]
Might be nice to be able to turn this off.

**Comment by shellscape** [comment]
@adam-lynch turn what off, specifically?

**Comment by adam-lynch** [comment]
@shellscape this error. I don't want to see it if it's not accurate :smile:

**Comment by shellscape** [comment]
@adam-lynch there's nothing we can do to hide or turn off that error, it's something thrown by gonzales-pe, and the module doesn't give us any information as to why it's thrown. It's typically a fault of incomplete support for LESS by the module (which is why we're looking at moving to a postcss parser).

**Comment by adam-lynch** [comment]
@shellscape ah ok :+1:

**Comment by samstr** [comment]
+1

this considers these `foreach` mixins invalid, despite them working fine:

https://github.com/seven-phases-max/less.curious/blob/master/articles/generic-for.md
https://github.com/seven-phases-max/less.curious/blob/master/articles/for-each.md

**Comment by shellscape** [comment]
You can track our progress on the switch to PostCSS on this issue: https://github.com/lesshint/lesshint/issues/158. We're close!

**Comment by ghaschel** [comment]
This also happens when using `& when` *inside* a mixin (less compiles it with no problems though)

```
webkit-border-radius: @radius;
-khtml-border-radius: @radius;
-moz-border-radius: @radius;
border-radius: @radius;
& when (@ie6 = true), (@ie7 = true), (@ie8 = true) {
	behavior: url('@{absPath}PIE.htc');
	-ms-behavior: url('@{absPath}PIE.htc');
}
```

**Comment by shellscape** [comment]
@ghaschel see my previous comment. these reports are moot until we switch to postcss

**Comment by ghaschel** [comment]
I saw your previous comment, I just thought it would be valid to point that the error also occurs when (via &) inside a mixin, and not just when defining it. =]

**Comment by afc163** [comment]
+1

**Comment by shellscape** [comment]
closing this as we've already referenced the issue that will fix these kinds of problems, and 2.0 will be testing soon.
